sycl/include/sycl/ext/intel/esimd/math.hpp

```diff
@@ -403,6 +403,38 @@ __ESIMD_UNARY_INTRINSIC_DEF(__ESIMD_EMATH_COND, sin, sin)
 /// Absolute error: \c 0.0008 or less for the range [-32767*pi, 32767*pi].
 __ESIMD_UNARY_INTRINSIC_DEF(__ESIMD_EMATH_COND, cos, cos)
 
+template <class T = double, int N, class Sat = saturation_off_tag>
+__ESIMD_API std::enable_if_t<std::is_same_v<T, double>, simd<double, N>>
+sqrt(simd<T, N> src, Sat sat = {}) {
+  return sqrt_ieee(src, sat);
```

Comment:
This line makes esimd::sqrt() an alias of esimd::sqrt_ieee, and I see some issue here.
Even though the GenX intrinsic `llvm.genx.sqrt` says it supports only `half and float`, it actually should support `double` type as well.
Please try using sqrt here. It will make the ESIMD implementation consistent with other functions, e.g. div - is fast, div_ieee is slower, but more precise/correct. Same for esimd::sqrt<float>: sqrt is faster while sqrt_ieee is more correct/precise.

If genx.sqrt really supports double (as said in IM), it would be possible to replace 2 functions added in this patch by simply modified condition, see few lines above:
```
/// Square root. Is not IEEE754-compatible.  Supports \c half and \c float.
/// Precision: 4 ULP.
__ESIMD_UNARY_INTRINSIC_DEF(__ESIMD_EMATH_COND, sqrt, sqrt)
```